packages/gatsby-source-contentful/src/fetch.js

```diff
@@ -1,11 +1,36 @@
 const contentful = require(`contentful`)
 const _ = require(`lodash`)
-
+const fs = require(`fs-extra`)
+const path = require(`path`)
 const normalize = require(`./normalize`)
 
-module.exports = async ({ spaceId, accessToken, host, syncToken, environment }) => {
+const cacheFilename = `contentful-result.json`
+
+async function writeCacheFile(dir, result) {
+  try {
+    await fs.writeJson(path.join(dir, cacheFilename), result)
+  } catch (err) {
+    console.error(err)
+  }
+}
+
+module.exports = async ({ spaceId, accessToken, host, syncToken, cacheDir, environment }) => {
   // Fetch articles.
   console.time(`Fetch Contentful data`)
+
+  // If the user knows they are offline, serve them cached result
+  if (process.env.GATSBY_CONTENTFUL_OFFLINE === `true`) {
```

Comment:
I think we should address very valid comment that is being removed below:
> // For prod builds though always fail if we can't get the latest data
and add check here to only try read cache if we are running `gatsby develop`. What do You think?

Reply:
Sound good to me how do we know that `gatsby develop` was used is there a command line var set?